rust/lance/src/io/exec/scalar_index.rs

```diff
@@ -452,6 +424,115 @@ impl MaterializeIndexExec {
     }
 }
 
+#[instrument(name = "make_row_ids", skip(mask, dataset, fragments))]
+async fn row_ids_for_mask(
+    mask: RowIdMask,
+    dataset: &Dataset,
+    fragments: &[Fragment],
+) -> Result<Vec<u64>> {
+    match (mask.allow_list, mask.block_list) {
+        (None, None) => {
+            // Matches all row ids in the given fragments.
+            if dataset.manifest.uses_move_stable_row_ids() {
+                let sequences = load_row_id_sequences(dataset, fragments)
+                    .map_ok(|(_frag_id, sequence)| sequence)
+                    .try_collect::<Vec<_>>()
+                    .await?;
+
+                let capacity = sequences.iter().map(|seq| seq.len() as usize).sum();
+                let mut row_ids = Vec::with_capacity(capacity);
+                for sequence in sequences {
+                    row_ids.extend(sequence.iter());
+                }
+                Ok(row_ids)
+            } else {
+                Ok(FragIdIter::new(fragments).collect::<Vec<_>>())
+            }
+        }
+        (Some(mut allow_list), None) => {
+            retain_fragments(&mut allow_list, fragments, dataset).await?;
+
+            if let Some(allow_list_iter) = allow_list.row_ids() {
+                Ok(allow_list_iter.map(u64::from).collect::<Vec<_>>())
```

Comment:
Why is `map(u64::from)` needed here?

Reply:
`.row_ids()` returns `RowAddress` while the return type is a u64. Should be a no-op when compiled.